tracking/include/pcl/tracking/impl/particle_filter.hpp

```diff
@@ -365,13 +365,11 @@ template <typename PointInT, typename StateT>
 void
 ParticleFilterTracker<PointInT, StateT>::update()
 {
-
   StateT orig_representative = representative_state_;
   representative_state_.zero();
   representative_state_.weight = 0.0;
-  for (const auto& p : *particles_) {
-    representative_state_ = representative_state_ + p * p.weight;
-  }
+  representative_state_ =
+      StateT::weightedAverage(particles_->points.begin(), particles_->points.end());
```

Comment:
```suggestion
      StateT::weightedAverage(particles_->begin(), particles_->end());
```

Reply:
@larshg thanks for the review.
Could you elaborate a bit on why to leave out points as this has been confusing me a bit since I feel like it is used interchangeably. 
For example in particle_filter.hpp they use particles->size() but also particles_->points.empty() or particles_->points.push_back().
I'm probably missing something obvious but could you perhaps give a short explanation on why this difference is there?

Reply:
Earlier on the .points has been used, since the other functions wasn't added, but its a bit of bad design to expose the internal data structure, so we are slowly progressing towards not using .points directly.
Mostly so, that it could be changed to other container types.

Reply:
Okay so if I understand correctly, the functionality is the same and the use of points is an old method that is slowly disappearing. The change you are proposing still does the same.